packages/protocol/test-sol/governance/voting/LockedGold.t.sol

```diff
@@ -0,0 +1,2890 @@
+  }
+
+  function test_ShouldAddPendingWithdrawals_WhenTheAccountIsRequestingOnlyNonVotingGold() public {
+    governance.setVoting(address(this));
+    governance.setTotalVotes(address(this), votingGold);
+
+    lockedGold.unlock(nonVotingGold);
+    (uint256[] memory vals, uint256[] memory timestamps) = lockedGold.getPendingWithdrawals(
+      address(this)
+    );
+    assertEq(vals.length, 1);
+    assertEq(timestamps.length, 1);
+    assertEq(vals[0], nonVotingGold);
+    assertEq(timestamps[0], availabilityTime);
+  }
+
+  function test_ShouldDecreaseTheACcountsNonVotingLockedGoldBalance_WhenTheAccountIsRequestingOnlyNonVotingGold()
+    public
+  {
+    governance.setVoting(address(this));
+    governance.setTotalVotes(address(this), votingGold);
+
+    lockedGold.unlock(nonVotingGold);
+    assertEq(lockedGold.getAccountNonvotingLockedGold(address(this)), votingGold);
+  }
+
+  function test_ShouldDecreaseTheAccountsTotalLockedGoldBalance_WhenTheAccountIsRequestingOnlyNonVotingGold()
+    public
+  {
+    governance.setVoting(address(this));
+    governance.setTotalVotes(address(this), votingGold);
+
+    lockedGold.unlock(nonVotingGold);
+    assertEq(lockedGold.getAccountTotalLockedGold(address(this)), votingGold);
+  }
+
+  function test_ShouldDecreaseTheNonVotingLockedGoldBalance_WhenTheAccountIsRequestingOnlyNonVotingGold()
+    public
+  {
+    governance.setVoting(address(this));
+    governance.setTotalVotes(address(this), votingGold);
+
+    lockedGold.unlock(nonVotingGold);
+    assertEq(lockedGold.getNonvotingLockedGold(), votingGold);
+  }
+
+  function test_ShouldDecreaseTheTotalLockedGoldBalance_WhenTheAccountIsRequestingOnlyNonVotingGold()
+    public
+  {
+    governance.setVoting(address(this));
+    governance.setTotalVotes(address(this), votingGold);
+
+    lockedGold.unlock(nonVotingGold);
+    assertEq(lockedGold.getTotalLockedGold(), votingGold);
+  }
+
+  function test_ShouldEmitGoldUnlockedEvent_WhenTheAccountIsRequestingOnlyNonVotingGold() public {
+    governance.setVoting(address(this));
+    governance.setTotalVotes(address(this), votingGold);
+
+    vm.expectEmit(true, true, true, true);
+    emit GoldUnlocked(address(this), nonVotingGold, availabilityTime);
+    lockedGold.unlock(nonVotingGold);
+  }
+
+  function test_ShouldRevert_WhenTheCorrectTimeIsEarlierThanTheRequirementTime_WhenThereIsBalanceRequirement()
+    public
+  {
+    validators.setAccountLockedGoldRequirement(address(this), balanceRequirement);
+    vm.expectRevert(
+      "Either account doesn't have enough locked Celo or locked Celo is being used for voting."
+    );
+    lockedGold.unlock(value);
+  }
+
+  function test_ShouldSucceed_WhenTheCorrectTimeIsEarlierThanTheRequirementTimeButRequestingCeloWithoutBalanceRequirement_WhenThereIsBalanceRequirement()
+    public
+  {
+    validators.setAccountLockedGoldRequirement(address(this), balanceRequirement);
+    lockedGold.unlock(value - balanceRequirement);
```

Comment:
worth adding a check that the number is actually right?

Reply:
I believe this test and previous test should be sufficient